### PR TITLE
CAS Ammo Loading Mismatch Bug Fix

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

Fixes a bug that allows players to load incorrect ammo into CAS weapons.

## Why It's Good For The Game

Fixes a bug.

Fixes: https://github.com/tgstation/TerraGov-Marine-Corps/issues/5828

## Changelog
:cl:
fix: Fixes a bug that allows players to load incorrect ammo into CAS weapons.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/dropship_equipment.dm [hunk 1]
```diff
@@ -167,6 +167,8 @@
 	var/screen_mode = 0
 	///how many points it costs to build this with the fabricator, set to 0 if unbuildable.
 	var/point_cost = 0
+	///what kind of ammo this uses if any
+	var/obj/structure/ship_ammo/ammo_type
```
Comment (TiviPlus) [inline code/game/objects/structures/dropship_equipment.dm:170]: A type

File: code/game/objects/structures/dropship_equipment.dm [hunk 2]
```diff
@@ -180,6 +182,8 @@
 		if(linked_console?.selected_equipment == src)
 			linked_console.selected_equipment = null
 		linked_console = null
+	if(ammo_type)
+		ammo_type = null //null the ammo type
```
Comment (TiviPlus) [inline code/game/objects/structures/dropship_equipment.dm:186]: Whitch is cast as a obj and nulled?
Don't do this

File: code/game/objects/structures/dropship_equipment.dm [hunk 3]
```diff
@@ -192,7 +196,7 @@
 		if(((!dropship_equipment_flags & IS_NOT_REMOVABLE) && !ship_base) || !(dropship_equipment_flags & USES_AMMO) || ammo_equipped || !istype(clamp.loaded, /obj/structure/ship_ammo))
 			return FALSE
 		var/obj/structure/ship_ammo/clamp_ammo = clamp.loaded
-		if(istype(type, clamp_ammo.equipment_type))
+		if(istype(type, clamp_ammo.equipment_type) || clamp_ammo.type != ammo_type) //Incompatible ammo
 			to_chat(user, "<span class='warning'>[clamp_ammo] doesn't fit in [src].</span>")
```
Comment (TiviPlus) [inline code/game/objects/structures/dropship_equipment.dm:196]: Strict typechecking this doesn't even work properly

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/dropship_equipment.dm [hunk 4]
```diff
@@ -167,6 +167,8 @@
 	var/screen_mode = 0
 	///how many points it costs to build this with the fabricator, set to 0 if unbuildable.
 	var/point_cost = 0
+	///what kind of ammo this uses if any
+	var/ammo_type
```
Comment (TiviPlus) [inline code/game/objects/structures/dropship_equipment.dm:172]: ammo_type_used

File: code/game/objects/structures/dropship_ammo.dm [hunk 5]
```diff
@@ -26,6 +26,8 @@
 	var/warning_sound = 'sound/machines/hydraulics_2.ogg'
 	var/ammo_used_per_firing = 1
 	var/point_cost = 0 //how many points it costs to build this with the fabricator, set to 0 if unbuildable.
+	///Type of ammo
+	var/ammo_type_used
```
Comment (TiviPlus) [inline code/game/objects/structures/dropship_ammo.dm:30]: Change it back on /obj/structure/ship_ammo I told you to do it BECAUSE they were named the same

Comment (Surrealaser) [reply code/game/objects/structures/dropship_ammo.dm:30]: Then say that and don't be cryptic; as a rule I don't have time to guess.

**Review by TiviPlus — APPROVED** [review]

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is now above zero (2). Feel free to make Feature/Tweak/Balance PRs.

**Pull request merged**